File: flies/RecipeDomain/Recipe.cs
using Flies.Interfaces;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace Flies.RecipeDomain;
public class Recipe : IDocument
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Id { get; set; }
    public string Name { get; set; }
    public Hook Hook { get; set; }
    public Thread Thread { get; set; }
    public Tool[] Tools { get; set; }
    public Supply[] Supplies { get; set; }
    public Instruction[] Instructions { get; set; }
    public Pattern Pattern { get; set; }

    public DateTime CreatedAt { get; set; }
    public DateTime ModifiedAt { get; set; }
}

File: frontend/src/app/fly/models/recipe.ts
import { Hook } from './hook';
import { Instruction } from './instruction';
import { Pattern } from './pattern';
import { Supply } from './supply';
import { Thread } from './thread';
import { Tool } from './tool';
export default interface Recipe {
  id: string;
  name: string;
  hook: Hook;
  thread: Thread;
  tools: Tool[];
  supplies: Supply[];
  instructions: Instruction[];
  pattern: Pattern;
  createdAt: string;
  modifiedAt: string;
}
